lib/Component/VCard.php

```diff
@@ -429,6 +435,67 @@ function jsonSerialize() {
     }
 
     /**
+     * This method serializes the data into XML. This is used to create xCard or
+     * xCal documents.
+     *
+     * @param Xml\Writer $writer  XML writer.
+     * @return void
+     */
+    function xmlSerialize(Xml\Writer $writer) {
+
+        $propertiesByGroup = [];
+
+        foreach ($this->children as $property) {
+
+            $group = $property->group;
+
+            if (!isset($propertiesByGroup[$group])) {
+                $propertiesByGroup[$group] = [];
+            }
+
+            $propertiesByGroup[$group][] = $property;
+
+        }
+
+        $writer->startElement(strtolower($this->name));
+
+        foreach ($propertiesByGroup as $group => $properties) {
+
+            if (!empty($group)) {
+
+                $writer->startElement('group');
+                $writer->writeAttribute('name', strtolower($group));
+
+            }
+
+            foreach ($properties as $property) {
+                switch ($property->name) {
+
+                    case 'VERSION':
+                        continue;
+
+                    case 'XML':
+                        $value = $property->getParts();
+                        $writer->writeRaw('  ' . $value[0] . "\n ");
```

Comment:
This looks a bit icky :) What about:

    $xmlFragment = new Sabre\Xml\Element\XmlFragment($value[0]);
    $writer->write($xmlFragment);